orderbook/src/main.rs

```diff
@@ -415,21 +415,35 @@ async fn main() {
     };
     check_database_connection(orderbook.as_ref()).await;
 
-    let serve_task = serve_task(
+    let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel();
+    let serve_api = serve_api(
         database.clone(),
         orderbook.clone(),
         fee_calculator,
         price_estimator,
         args.bind_address,
+        shutdown_receiver,
```

Comment:
In order to make `shutdown_receiver` a `impl Future<Output = ()>` you can:
```suggestion
        async { shutdown_receiver; },
```